src/apps/Settings/HomeSettings.js

```diff
@@ -0,0 +1,169 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { Button, DropDown, Field, TextInput } from '@aragon/ui'
+
+import { AppType, EthereumAddressType } from '../../prop-types'
+import AragonStorage from '../../storage/storage-wrapper'
+
+import Option from './Option'
+
+const defaultName = 'Use the default'
+
+class HomeSettings extends React.Component {
+  static propTypes = {
+    account: EthereumAddressType,
+    apps: PropTypes.arrayOf(AppType).isRequired,
+    walletWeb3: PropTypes.object.isRequired,
+  }
+  state = {
+    homeAppName: 'Home',
+    selectedHomeAppName: defaultName,
+    storageApp: null,
+  }
+
+  async getHomeSettings(props) {
+    const { apps, walletWeb3, account } = props
+    const storageApp = apps.find(({ name }) => name === 'Storage')
+    let homeAppName, homeAppAddr
+
+    // TODO: Move this code to a subscriber
+    if (storageApp && storageApp.proxyAddress) {
+      homeAppAddr = await AragonStorage.get(
+        walletWeb3,
+        storageApp.proxyAddress,
+        account,
+        'HOME_APP'
+      )
+
+      homeAppName = await AragonStorage.get(
+        walletWeb3,
+        storageApp.proxyAddress,
+        account,
+        'HOME_APP_NAME'
+      )
+      if (!homeAppName) {
+        homeAppName = 'Home'
+      }
+    }
+    const selectedHomeApp = apps.find(
+      ({ proxyAddress }) => proxyAddress === homeAppAddr
+    )
+
+    this.setState({
+      homeAppName: homeAppName,
+      selectedHomeAppName:
+        (selectedHomeApp && selectedHomeApp.name) || defaultName,
+      storageApp: apps.find(({ name }) => name === 'Storage'),
+    })
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.getHomeSettings(nextProps)
+  }
+  componentWillMount() {
+    this.getHomeSettings(this.props)
+  }
+
+  handleHomeAppChange = (index, apps) => {
+    this.setState({ selectedHomeAppName: apps[index] })
+  }
+
+  handleHomeNameChange = event => {
+    this.setState({
+      homeAppName: event.target.value && event.target.value.trim(),
+    })
+  }
+
+  handleHomeSettingsSave = async () => {
+    const { walletWeb3, apps, account } = this.props
+    const { storageApp, selectedHomeAppName, homeAppName } = this.state
+
+    if (storageApp && storageApp.proxyAddress) {
+      const selectedAppAddr = apps.find(
+        ({ name }) => name === selectedHomeAppName
+      )
+
+      try {
+        if (selectedHomeAppName !== defaultName) {
+          await AragonStorage.set(
+            walletWeb3,
+            storageApp.proxyAddress,
+            account,
+            'HOME_APP_NAME',
+            homeAppName
+          )
+        }
+
+        await AragonStorage.set(
```

Comment:
this triggers 2 transactions, can this transaction be avoided?

Reply:
Also, the signer panel is never shown, It just directly shows up the metamask transaction window, because the Wrapper.js should be used to handle the transaction.

Reply:
We can check if the next value is the same that current value and avoid that transaction. But based on the contract specs if you change the home app, and it's name, you will have to sign 2 transactions.